components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/CachedBackedConfigurationDAO.java

```diff
@@ -403,4 +404,44 @@ private void addResourceToCache(Resource resource) throws ConfigurationManagemen
             PrivilegedCarbonContext.endTenantFlow();
         }
     }
+
+    private void deleteResourceFromCache(Resource resource) throws ConfigurationManagementException {
+
+        if (resource == null) {
+            return;
+        }
+
+        int tenantId;
+        try {
+            tenantId = ConfigurationManagerComponentDataHolder.getInstance().getRealmService()
+                    .getTenantManager().getTenantId(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+        } catch (UserStoreException e) {
+            throw new ConfigurationManagementException("Error when setting tenant domain. ",
+                    ConfigurationConstants.ErrorMessages.ERROR_CODE_UNEXPECTED.getCode(), e);
+        }
+
+        try {
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                    .setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+
+            ResourceByIdCacheKey resourceByIdCacheKey = new ResourceByIdCacheKey(resource.getResourceId(),
+                    resource.getTenantDomain());
+            ResourceByNameCacheKey resourceByNameCacheKey = new ResourceByNameCacheKey(resource.getResourceName(),
+                    resource.getTenantDomain());
+
+            if (log.isDebugEnabled()) {
+                String message = String.format("Following two cache entries created. 1. Resource by name cache %s, 2." +
```

Comment:
Isn't this log entry incorrect? We are clearing the cache here.

Reply:
@JKAUSHALYA Sorry. Fixed.